src/app/camo-card/camo-card.component.ts

```diff
@@ -9,7 +8,7 @@ import { ProgressData } from '../home-page/ProgressData';
 })
 export class CamoCardComponent implements OnInit {
   @Input() weapon: Weapon;
-  @Input() progress: ProgressData;
+  @Input() progress: Progress;
```

Comment:
CamoCard doesn't take ProgressData, it shouldn't care about progress of other weapons. It cares only about the current weapon's progress, so it should only take the Progress and you pass it in by `<app-camo-card [progress]='progressData[weapon.id]'></app-camo-card>`